app/helpers/otus/catalog_helper.rb

```diff
@@ -65,9 +65,17 @@ def otu_descendants_and_synonyms(
     similar_otus: [],
     common_names: false,
     language_alpha2: nil,
-    max_descendants_depth: Float::INFINITY
+    max_descendants_depth: Float::INFINITY,
+    project_scope_tag_id: nil
   )
-    s = Otu.where(taxon_name_id: otu.taxon_name_id).where.not(id: otu.id).where.not(name: (otu.name.presence)).to_a
+    q = Otu.where(taxon_name_id: otu.taxon_name_id).where.not(id: otu.id).where.not(name: (otu.name.presence))
+
+    if project_scope_tag_id.present?
+      q = q.joins(:tags)
+        .where("tags.tag_object_type = 'Otu' AND tags.keyword_id = #{project_scope_tag_id}")
```

Comment:
I don't need `tags.tag_object_type = 'Otu'` here, it's covered by `joins(:tags)`